pkg/api/controllers/catalog/catalog.go

```diff
@@ -21,21 +23,102 @@ type CacheCleaner struct {
 	catalogClient        v3.CatalogInterface
 	projectCatalogClient v3.ProjectCatalogInterface
 	clusterCatalogClient v3.ClusterCatalogInterface
+	manager              *manager.Manager
 	debounce             func(func())
 }
 
 func Register(ctx context.Context, context *config.ScaledContext) {
+	managementCtx, err := context.NewManagementContext()
+	if err != nil {
+		panic(err)
+	}
 	cleaner := &CacheCleaner{
 		catalogClient:        context.Management.Catalogs(""),
 		projectCatalogClient: context.Management.ProjectCatalogs(""),
 		clusterCatalogClient: context.Management.ClusterCatalogs(""),
+		manager:              manager.New(managementCtx),
 		debounce:             debounce.New(time.Minute),
 	}
 	go cleaner.runPeriodicCatalogCacheCleaner(ctx, time.Hour)
 
 	context.Management.Catalogs("").Controller().AddHandler(ctx, "catalogCache", cleaner.destroyCatalogSync)
 	context.Management.ClusterCatalogs("").Controller().AddHandler(ctx, "clusterCatalogCache", cleaner.destroyClusterCatalogSync)
 	context.Management.ProjectCatalogs("").Controller().AddHandler(ctx, "projectCatalogCache", cleaner.destroyProjectCatalogSync)
+
+	// Don't sync catalog templates on leader node of duplicate actions
+	if context.PeerManager != nil && !context.PeerManager.IsLeader() {
+		logrus.Info("registering scaled context catalog handler")
+		context.Management.Catalogs("").Controller().AddHandler(ctx, "scaledCatalogSync", cleaner.scaledCatalogSync)
+		context.Management.ClusterCatalogs("").Controller().AddHandler(ctx, "scaledClusterCatalogSync", cleaner.scaledClusterCatalogSync)
+		context.Management.ProjectCatalogs("").Controller().AddHandler(ctx, "scaledProjectCatalogSync", cleaner.scaledProjectCatalogSync)
+	}
+}
+
+func (c *CacheCleaner) scaledCatalogSync(key string, obj *v3.Catalog) (runtime.Object, error) {
+	if obj == nil {
+		return nil, nil
+	}
+
+	// always get a refresh catalog from etcd
+	catalog, err := c.catalogClient.Get(key, metav1.GetOptions{})
```

Comment:
Global catalogs are also namespaced and are created under the cattle-global-data namespace, shouldn't we be using GetNamespaced for those too?

Reply:
updated.